R/read_zarr.R

```diff
@@ -0,0 +1,47 @@
+#' Read Zarr
+#'
+#' Read data from a Zarr store
+#'
+#' @param path Path to the Zarr store to read
+#' @param to The type of object to return. Must be one of: "InMemoryAnnData",
+#'   "HDF5AnnData", "SingleCellExperiment", "Seurat"
+#' @param ... Extra arguments provided to [to_SingleCellExperiment()] or
+#'   [to_Seurat()]
+#'
+#' @return The object specified by `to`
+#' @export
+#'
+#' @examples
+#' file <- system.file("extdata", "example.zarr", package = "anndataR")
+#' store <- pizzarr::DirectoryStore$new(file)
+#'
+#' # Read the Zarr store as a SingleCellExperiment object
+#' if (requireNamespace("SingleCellExperiment", quietly = TRUE)) {
+#'   sce <- read_zarr(store, to = "SingleCellExperiment")
+#' }
+#'
+#' # Read the Zarr store as a Seurat object
+#' if (requireNamespace("SeuratObject", quietly = TRUE)) {
+#'   seurat <- read_zarr(store, to = "Seurat")
+#' }
+read_zarr <- function(
+    path,
+    to = c("InMemoryAnnData", "ZarrAnnData", "SingleCellExperiment", "Seurat"),
+    ...) {
+  to <- match.arg(to)
+
+  adata <- ZarrAnnData$new(path)
+
+  fun <- switch(to,
+    "SingleCellExperiment" = to_SingleCellExperiment,
+    "Seurat" = to_Seurat,
+    "InMemoryAnnData" = to_InMemoryAnnData,
+    "ZarrAnnData" = NULL
+  )
```

Comment:
Should add HDF5AnnData as an option here as well. Which means that we should probably store this information elsewhere to avoid code duplication

```suggestion
read_zarr <- function(
    path,
    to = c("InMemoryAnnData", "HDF5AnnData", "SingleCellExperiment", "Seurat", "ZarrAnnData"),
    ...) {
  to <- match.arg(to)

  adata <- ZarrAnnData$new(path)

  fun <- switch(to,
    "SingleCellExperiment" = to_SingleCellExperiment,
    "Seurat" = to_Seurat,
    "InMemoryAnnData" = to_InMemoryAnnData,
    "HDF5AnnData" = to_HDF5AnnData,
    "ZarrAnnData" = NULL
  )

Reply:
done ... mixed this with the latest commit! https://github.com/scverse/anndataR/pull/190/commits/d192e68ed312f212476a5490c973619f08e7c8de